### PR TITLE
Ignore fastboot build output file if app unused/disable fastboot

### PR DESCRIPTION
**Pull request by houfeng0923**
hi @xg-wang :
i am doing work on reducing assets files size for deploy.
there is no ember-cli-fastboot in my project , so can we remove fetch-fastboot file in outputTree ? .

### CODE REVIEW

**Comment by houfeng0923** [comment]
btw, i find a related issue : https://github.com/ember-cli/ember-fetch/issues/320

File: index.js [hunk 1]
```diff
@@ -141,7 +141,10 @@ module.exports = {
 
   // Only include public/fetch-fastboot.js if top level addon
   treeForPublic() {
-    return !this.parent.parent ? this._super.treeForPublic.apply(this, arguments) : null;
+    const fastbootEnabled = process.env.FASTBOOT_DISABLED !== 'true' 
+      && !!this.project.findAddonByName('ember-cli-fastboot');
+    return !this.parent.parent && fastbootEnabled;
```
Comment (xg-wang) [inline index.js]: The `;` failed tests

Comment (houfeng0923) [reply index.js]: ah, fixed it

**Comment by houfeng0923** [comment]
hi, @xg-wang , unit test appended is passed in local dev env , but CI is fail in pull.

**Comment by xg-wang** [comment]
@houfeng0923 I think CI fails because node8 doesn't have `.finally`

**Comment by houfeng0923** [comment]
sorry, @xg-wang,  last commit is passed in unit test, but CI always fail ,
maybe we need upgrade dependencies

File: test/without-fastboot-build-test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  });
+
+  it('builds no exist dist/ember-fetch/fetch-fastboot.js', function() {
+    return app.runEmberCommand('build').then(function() {
+      expect(app.filePath('dist/index.html')).to.be.a.file();
+      expect(app.filePath('dist/ember-fetch')).to.not.be.a.path();
+    });
+  });
+
+  it('build with process.env.FASTBOOT_DISABLED', function() {
```
Comment (xg-wang) [inline test/without-fastboot-build-test.js]: The default app does not have ember-cli-fastboot, this test needs to install then set the flag to actually test the case.

**Comment by xg-wang** [comment]
@houfeng0923 the dependency issue is resolved, I tried to push your branch to a fork and ci is passing https://github.com/xg-wang/ember-fetch/runs/533697154

File: test/without-fastboot-build-test.js (hunk 2 above)
Comment (houfeng0923) [reply test/without-fastboot-build-test.js]: i know u meaning, but i think it's a case point in ember fastboot.
but if you think so, case can update  .

**Pull request merged**